framework/src/base/UpdateDisplacedMeshThread.C

```diff
@@ -13,48 +13,56 @@
 /****************************************************************/
 
 #include "UpdateDisplacedMeshThread.h"
+
+#include "AllNodesSendListThread.h"
 #include "DisplacedProblem.h"
 #include "MooseMesh.h"
-#include "SubProblem.h"
+
+#include "libmesh/numeric_vector.h"
 
 UpdateDisplacedMeshThread::UpdateDisplacedMeshThread(FEProblemBase & fe_problem,
                                                      DisplacedProblem & displaced_problem)
-  : ThreadedNodeLoop<SemiLocalNodeRange, SemiLocalNodeRange::const_iterator>(fe_problem),
+  : ThreadedNodeLoop<NodeRange, NodeRange::const_iterator>(fe_problem),
     _displaced_problem(displaced_problem),
     _ref_mesh(_displaced_problem.refMesh()),
     _nl_soln(*_displaced_problem._nl_solution),
     _aux_soln(*_displaced_problem._aux_solution),
+    _nl_ghosted_soln(NumericVector<Number>::build(_nl_soln.comm()).release()),
```

Comment:
Why release this only to capture it in a different smart pointer? Unless there's a compelling reason, I'd just keep the `unique_ptr` around from the build.